### PR TITLE
Pandasway

### PR DESCRIPTION
**Pull request by Tompoudurans**
added SQL data functions using pandas

(i will add more docstrings and tests as well as reformatting later on)

### CODE REVIEW

File: src/dataexplorer/__main__.py [hunk 1]
```diff
@@ -1,15 +1,144 @@
...
+    """
+    provides the basic stats for the dataset
+    """
+    mean = data.mean()
+    std = data.std()
+    maxi = data.max()
+    mini = data.min()
+    no_field = len(data.columns)
+    length = len(data)
+    return [mean, std, maxi, mini, no_field, length]
```
Comment (drvinceknight) [inline src/dataexplorer/__main__.py:40]:
```suggestion
    provides the basic stats for the dataset
    """
    return data.describe()
```

File: src/dataexplorer/__main__.py [hunk 2]
```diff
@@ -1,15 +1,144 @@
...
+
+
+def categorical_handle(data):
```
Comment (drvinceknight) [inline src/dataexplorer/__main__.py:43]: I don't really know what this function is trying to do. You might be wanting to do encoding, this blog post has some good information on how to do it. I might be misunderstanding however in which case can you expand on the docstring please.

File: src/dataexplorer/__main__.py [hunk 3]
```diff
@@ -1,15 +1,144 @@
...
+
+
+def place_time(data, cat):
+    """
+    transform time from numerical back to ctime
+    """
+    line = data[cat]
+    fulltime = []
+    for l in line:
+        print(l)
```
Comment (drvinceknight) [inline src/dataexplorer/__main__.py:70]:
```suggestion

```

File: src/dataexplorer/__main__.py [hunk 4]
```diff
@@ -1,15 +1,144 @@
...
+        elif "<M8[ns]" == data[name].dtype:
+            times = timestamp_handle(data[name])
+            data[name] = times
+            indexs.append(["time"])
+        else:
+            indexs.append(["#"])
+    return data, indexs
+
+
+def place_time(data, cat):
```
Comment (drvinceknight) [inline src/dataexplorer/__main__.py:63]: Why is this necessary? The idea of the explore functionality is to not go in to details like this.

File: src/dataexplorer/__main__.py [hunk 5]
```diff
@@ -1,15 +1,144 @@
...
+    return data.columns
+
+
+def place_columns(database, col):
+    """
+    insert column name to the dataset
+    """
+    database.columns = col
+    return database
+
```
Comment (drvinceknight) [inline src/dataexplorer/__main__.py:89]: I don't understand why we need these.

File: src/dataexplorer/__main__.py [hunk 6]
```diff
@@ -1,15 +1,144 @@
...
+def main_read(target, table, output):
+    """
+    read data and logs it
+    """
+    data = read_sql(target,table)
+    stats = get_stats(data)
+    cloumn = collect_column(data)
+    numeric,index_list = categorical_handle(data)
+    make_log_file([stats,cloumn,index_list], output)
+    export_numeric(numeric)
```
Comment (drvinceknight) [inline src/dataexplorer/__main__.py:135]: You can get rid of a lot of code you've written and just go with: 

```suggestion
    return data.describe()
```

File: src/dataexplorer/__main__.py (hunk 5 above)
Comment (Tompoudurans) [reply src/dataexplorer/__main__.py:89]: I have slightly miss understood the purpose of the project, I thought it was going to be a copy of the pre-processing used for GAN

File: src/dataexplorer/__main__.py (hunk 1 above)
Comment (Tompoudurans) [reply src/dataexplorer/__main__.py:40]: I did not know you could do it like this

**Comment by drvinceknight** [comment]
This is looking better, could you add some tests of the cli tool perhaps (I'm not convinced it's working as supposed to - but that might just be my `click` knowledge being a bit rusty - tests will clarify this). Here is an example of some tests for command line interfaces: https://github.com/drvinceknight/nbchkr/blob/dev/tests/test_cli.py

As you'll see there I'm essentially using the tool as expected and capturing the output. I make use of the `subprocess` tool to do that.

**Comment by Tompoudurans** [comment]
note the commits above is me rebasing pandasway

**Review by drvinceknight — APPROVED** [review]
This looks good. :+1:
